exchanges/kraken/kraken_websocket.go

```diff
@@ -214,6 +254,192 @@ func (k *Kraken) WsHandleEventResponse(response *WebsocketEventResponse, rawResp
 	}
 }
 
+func (k *Kraken) wsHandleAuthDataResponse(response WebsocketDataResponse) {
+	if chName, ok := response[1].(string); ok {
+		switch chName {
+		case krakenWsOwnTrades:
+			if k.Verbose {
+				log.Debugf(log.ExchangeSys, "%v Websocket auth own trade data received",
+					k.Name)
+			}
+			k.wsProcessOwnTrades(&response[0])
+		case krakenWsOpenOrders:
+			if k.Verbose {
+				log.Debugf(log.ExchangeSys, "%v Websocket auth open order data received",
+					k.Name)
+			}
+			k.wsProcessOpenOrders(&response[0])
+		}
+	}
+}
+
+func (k *Kraken) wsProcessOwnTrades(ownOrders interface{}) {
+	if data, ok := ownOrders.([]interface{}); ok {
+		for i := range data {
+			ownTrade := data[i].(map[string]interface{})
+			for _, val := range ownTrade {
+				tradeData := val.(map[string]interface{})
+				cost, err := strconv.ParseFloat(tradeData["cost"].(string), 64)
+				if err != nil {
+					k.Websocket.DataHandler <- err
+				}
+				fee, err := strconv.ParseFloat(tradeData["fee"].(string), 64)
+				if err != nil {
+					k.Websocket.DataHandler <- err
+				}
+				margin, err := strconv.ParseFloat(tradeData["margin"].(string), 64)
+				if err != nil {
+					k.Websocket.DataHandler <- err
+				}
+				vol, err := strconv.ParseFloat(tradeData["vol"].(string), 64)
+				if err != nil {
+					k.Websocket.DataHandler <- err
+				}
+				price, err := strconv.ParseFloat(tradeData["price"].(string), 64)
+				if err != nil {
+					k.Websocket.DataHandler <- err
+				}
+				timeTogether, err := strconv.ParseFloat(tradeData["time"].(string), 64)
+				if err != nil {
+					k.Websocket.DataHandler <- err
+				}
+				first, second, err := convert.SplitFloatDecimals(timeTogether)
+				if err != nil {
+					k.Websocket.DataHandler <- err
+				}
+				k.Websocket.DataHandler <- WsOwnTrade{
+					Cost:               cost,
+					Fee:                fee,
+					Margin:             margin,
+					OrderTransactionID: tradeData["ordertxid"].(string),
+					OrderType:          tradeData["ordertype"].(string),
+					Pair:               tradeData["pair"].(string),
+					PostTransactionID:  tradeData["postxid"].(string),
+					Price:              price,
+					Time:               time.Unix(first, second),
+					Type:               tradeData["type"].(string),
+					Vol:                vol,
+				}
+			}
+		}
+	} else {
+		k.Websocket.DataHandler <- errors.New(k.Name + " - Invalid own trades data")
+	}
+}
+
+func (k *Kraken) wsProcessOpenOrders(ownOrders interface{}) {
+	if data, ok := ownOrders.([]interface{}); ok {
+		for i := range data {
+			ownTrade := data[i].(map[string]interface{})
+			for key, val := range ownTrade {
+				tradeData := val.(map[string]interface{})
+				if len(tradeData) == 1 {
+					// just a status update
+					if status, ok := tradeData["status"].(string); ok {
+						k.Websocket.DataHandler <- "Order " + key + " " + status
```

Comment:
Please add k.Name here like what you've done with the errors sent to the data handler